src/newViews/PillarAnalysis/AnalyticalStatementInput/AnalyticalNGramsModal/index.tsx

```diff
@@ -0,0 +1,298 @@
+    mapToList,
+} from '@togglecorp/fujs';
+import {
+    TextArea,
+    PendingMessage,
+    Modal,
+    Button,
+    ListView,
+    Container,
+} from '@the-deep/deep-ui';
+import {
+    ResponsiveContainer,
+    BarChart,
+    Bar,
+    XAxis,
+    YAxis,
+    Tooltip,
+    Legend,
+} from 'recharts';
+
+import { useRequest } from '#utils/request';
+import ExcerptOutput from '#newComponents/viewer/ExcerptOutput';
+
+import {
+    PartialAnalyticalStatementType,
+} from '../../schema';
+import EntryContext, { EntryFieldsMin } from '../../context';
+
+import styles from './styles.scss';
+
+const chartMargins = {
+    top: 0,
+    bottom: 0,
+    right: 10,
+    left: 10,
+};
+
+interface NgramsResponse {
+    ngrams: {
+        unigrams: {
+            [key in string]: number;
+        };
+        bigrams: {
+            [key in string]: number;
+        };
+        trigrams: {
+            [key in string]: number;
+        };
+    }
+}
+
+const keySelector = (item: EntryFieldsMin) => item.id;
+
+interface Props {
+    onModalClose: () => void,
+    statementId: string;
+    mainStatement: string | undefined,
+    onStatementChange: (newVal: string | undefined) => void;
+    analyticalEntries: PartialAnalyticalStatementType['analyticalEntries']
+    onNgramClick: (item: string) => void;
+}
+
+function AnalyticalNGramsModal(props: Props) {
+    const {
+        onModalClose,
+        mainStatement,
+        onStatementChange,
+        statementId,
+        analyticalEntries,
+        onNgramClick,
+    } = props;
+
+    const { entries } = useContext(EntryContext);
+    const [tempMainStatement, setTempMainStatement] = useState<string | undefined>(mainStatement);
+    const [pristine, setPristine] = useState(true);
+
+    const handleNgramClick = useCallback((item) => {
+        const typedItem = item as { name: string };
+        onNgramClick(typedItem?.name);
+        onModalClose();
+    }, [onNgramClick, onModalClose]);
+
+    const entriesForNgrams = useMemo(() => (
+        analyticalEntries?.map(
+            ae => (ae.entry ? entries?.[ae.entry] : undefined),
+        ).filter(isDefined) ?? []
+    ), [entries, analyticalEntries]);
+
+    const entryPayload = useMemo(() => {
+        const excerptTexts = entriesForNgrams.map(ae => ae.excerpt);
+
+        return ({
+            entries: excerptTexts.filter(ae => ae && ae.length > 3).filter(isDefined),
+            ngrams: {
+                unigrams: true,
+                bigrams: true,
+                trigrams: true,
+            },
+            enable_stopwords: true,
+            enable_stemming: false,
+            enable_case_sensitive: true,
+            max_ngrams_items: 10,
+        });
+    }, [entriesForNgrams]);
+
+    const {
+        pending,
+        response: ngramsResponse,
+    } = useRequest<NgramsResponse>({
+        url: 'serverless://ngram-process/',
+        method: 'POST',
+        body: entryPayload,
+        failureHeader: 'Failed to get ngrams',
+    });
```

Comment:
Should use GET request to get data from server

Reply:
Since the API info requires POST request and also requires payload to be sent, POST request has been implemented.